src/client/utils/hooks/useQuoteCartData.ts

```diff
@@ -1,21 +1,218 @@
 import { useParams } from 'react-router-dom'
 import { useQuoteCartQuery } from 'data/graphql'
 import { useCurrentLocale } from 'l10n/useCurrentLocale'
+import { getMainQuote } from 'api/quoteBundleSelectors'
+import { getSelectedBundleVariant } from 'api/quoteCartQuerySelectors'
+import { PriceData } from 'src/client/pages/Checkout/shared/types'
+import {
+  typeOfResidenceTextKeys,
+  HomeInsuranceTypeOfContract,
+} from 'pages/OfferNew/utils'
+
+import {
+  GenericQuoteData,
+  HouseQuoteDetails,
+} from 'src/client/api/quoteDetailsDataTypes'
+import { isHouseQuoteDetailsData } from 'api/quoteDetailsDataSelectors'
+import { formatPostalNumber } from '../postalNumbers'
+import { useSelectedInsuranceTypes } from './useSelectedInsuranceTypes'
+
+export type Value = {
+  value?: string | number
+  textKey?: string
+  prefix?: string
+  suffix?: string
+}
+
+type Details = {
+  label: string
+  value: Value
+}
+
+type DetailsGroup = Details[]
 
 export const useQuoteCartData = () => {
   const { id: quoteCartId } = useParams<{ id: string }>()
   const { isoLocale } = useCurrentLocale()
-
   const { data, loading, error } = useQuoteCartQuery({
-    variables: {
-      id: quoteCartId,
-      locale: isoLocale,
-    },
+    variables: { id: quoteCartId, locale: isoLocale },
+  })
+
+  const [selectedInsuranceTypes] = useSelectedInsuranceTypes()
+  const selectedQuoteBundle = getSelectedBundleVariant(
+    data,
+    selectedInsuranceTypes,
+  )
+  if (!selectedQuoteBundle) return null
+
+  const prices = selectedQuoteBundle?.bundle.quotes.map((item) => {
+    return { displayName: item.displayName, price: item.price.amount }
   })
 
+  const mainQuote = getMainQuote(selectedQuoteBundle.bundle)
+
+  const getPriceData = (): PriceData => {
+    return {
+      prices: prices,
+      discount: selectedQuoteBundle.bundle.bundleCost.monthlyDiscount.amount,
```

Comment:
sure, I can do something like this. 
`export const getDiscountAmount = (bundle: QuoteBundle) => {
  return bundle.bundleCost.monthlyDiscount.amount
}`

I do think that kind of adds some more complexity to the file as we add more lines and instead of maybe directly getting a value from the scope we do a function hosted in another file 🤔 . what do you think ?

Reply:
The idea behind the selectors was to have one place that is tightly coupled to the GQL api and that the components are API agnostic as possible. Getting the amount is something we do in a lot of other places, so I am sure this function can/will be reused.